src/jdk.attach/aix/classes/sun/tools/attach/VirtualMachineImpl.java

```diff
@@ -57,14 +56,9 @@ public class VirtualMachineImpl extends HotSpotVirtualMachine {
         super(provider, vmid);
 
         // This provider only understands pids
-        int pid;
-        try {
-            pid = Integer.parseInt(vmid);
-            if (pid < 1) {
-                throw new NumberFormatException();
-            }
-        } catch (NumberFormatException x) {
-            throw new AttachNotSupportedException("Invalid process identifier: " + vmid);
+        int pid = Integer.parseInt(vmid);
+        if (pid < 1) {
+            throw new AttachNotSupportedException("Invalid process identifier -1");
```

Comment:
Why the change to using -1 instead of `vmid`? It seems presumptuous that it is -1, and also seems less useful than printing `vmid`.